Source/Core/DolphinQt/DiscordHandler.h

```diff
@@ -0,0 +1,43 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+// Note using a ifdef around this class causes link issues with qt
+
+#include <list>
+#include <thread>
+
+#include <QObject>
+
+#include "Common/Flag.h"
+
+#include "UICommon/DiscordPresence.h"
+
+class DiscordJoinRequestDialog;
+
+class DiscordHandler : public QObject, public Discord::Handler
+{
+  Q_OBJECT
+public:
+  explicit DiscordHandler(QWidget* parent);
+  ~DiscordHandler();
+
+  void Start();
+  void Stop();
+  void DiscordJoin() override;
+  void DiscordJoinRequest(const char* id, const std::string& discord_tag,
+                          const char* avatar) override;
+  void ShowNewJoinRequest(const char* id, const std::string discord_tag, const char* avatar);
+signals:
+  void Join();
+  void JoinRequest(const char* id, const std::string discord_tag, const char* avatar);
```

Comment:
We need to use `std::string` here, as `const char*` is just a pointer and not a copy.